src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -80,11 +81,37 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         validateStreamId(cursors, streamId, subscriptionId);
 
+        LOG.debug("[COMMIT_CURSORS] stream IDs validation finished");
+
+        final Iterator<EventTypePartition> distinctPartitions = cursors.stream()
```

Comment:
Yes I wanted to do it like that in the beginning , in this case the implementation would be much more simpler. But the tricky point here is that we need to return response for each cursor. So if I would just commit the max and will ignore the others I will not know which result to return for each cursor. For example if user provides these cursors for a partition: `24, 25, 27, 10, 35, 30` we should respond with `true, true, true, false, true, false`

In this test you can see which cases are possible https://github.com/zalando/nakadi/pull/563/files#diff-d92c44fce8107aa3413fffc22d08eed9R175